### PR TITLE
[4.0] Adapting privacycheck quickicon to 4.0

### PR DESCRIPTION
**Pull request by infograf768**
### Testing Instructions
Display Control panel.

### Before patch
Screen Shot 2019-04-09 at 10 28 47

### After patch
Screen Shot 2019-04-09 at 10 29 07

@wilsonge

### CODE REVIEW

**Comment by ghost** [comment]
Test by Patchtester or npm?

**Comment by infograf768** [comment]
patchtester should work here as the js has been added a while ago.

**Comment by infograf768** [comment]
BUT some aspects have just been merged today:
https://github.com/joomla/joomla-cms/pull/24531

Therfeore it may have to be tested with that patch in when clicking on the icon

**Comment by ghost** [comment]
~~I have tested this item :white_check_mark: successfully on 7200c0f9a79c906d616e21a40551ec3a09b57b32~~<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24534">issues.joomla.org/tracker/joomla-cms/24534</a>.</sub>

**Comment by ghost** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24534">issues.joomla.org/tracker/joomla-cms/24534</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 7200c0f9a79c906d616e21a40551ec3a09b57b32<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24534">issues.joomla.org/tracker/joomla-cms/24534</a>.</sub>

**Comment by infograf768** [comment]
restarting drone but it may concern rips
@zero24

File: plugins/quickicon/privacycheck/privacycheck.php [hunk 1]
```diff
@@ -3,14 +3,16 @@
...
  * @subpackage  Quickicon.privacycheck
  *
- * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @copyright   Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
 defined('_JEXEC') or die;
 
 use Joomla\CMS\Factory;
```
Comment (SharkyKZ) [inline plugins/quickicon/privacycheck/privacycheck.php]: Remove this. It's not needed anymore.

File: plugins/quickicon/privacycheck/privacycheck.php [hunk 2]
```diff
@@ -29,6 +31,14 @@ class PlgQuickiconPrivacyCheck extends JPlugin
 	 */
 	protected $autoloadLanguage = true;
 
+	/**
+	 * Application object.
+	 *
+	 * @var    \Joomla\CMS\Application\CMSApplication
+	 * @since  3.7.0
```
Comment (SharkyKZ) [inline plugins/quickicon/privacycheck/privacycheck.php]: Use `__DEPLOY_VERSION__`?

Comment (infograf768) [reply plugins/quickicon/privacycheck/privacycheck.php]: will do

File: plugins/quickicon/privacycheck/privacycheck.php (hunk 1 above)
Comment (infograf768) [reply plugins/quickicon/privacycheck/privacycheck.php]: Indeed. Looks like the other quickicon plugins also still have `use Joomla\CMS\Factory;`

**Comment by infograf768** [comment]
@SharkyKZ 
Just to remember that anyway we will have to modify this plugin as its js `privacycheck.js`  uses JQuery and we have to modify this.
Therefore this is just a temporary fix until someone modifies both on the model of the other quickicons js and php

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 2f59918b538cb501a3a534bc606b610d2819a6a6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24534">issues.joomla.org/tracker/joomla-cms/24534</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**